### PR TITLE
fix: loosing block settings when adding new block

### PR DESCRIPTION
**Pull request by nileshgulia1**
addresses: #2330

### CODE REVIEW

**Comment by nileshgulia1** [comment]
@sneridagh @tiberiuichim In the above scenario, we're "adding" another block right after the current one instead of mutation.  I intend to move the 'block adding' logic inside the addBlock helper. 
Add an extra text block after adding a block (let's say image) from the last one.

I also propose another solution by keeping the block mutation intact by just copying the existing block settings into the new block.

**Comment by avoinea** [comment]
@nileshgulia1 We should try not to change the current behavior and insert new blocks before the Text one.

**Comment by nileshgulia1** [comment]
Demo: 


https://user-images.githubusercontent.com/22280901/113610901-5ebc5500-966b-11eb-8c37-5d0d7b338f1c.mov

**Comment by avoinea** [comment]
Duplicate #2337

File: src/helpers/Blocks/Blocks.js [hunk 1]
```diff
@@ -160,31 +160,39 @@ export function mutateBlock(formData, id, value) {
...
+    return {
+      ...formData,
+      [blocksFieldname]: {
+        ...formData[blocksFieldname],
+        [idnewBlock]: value || null,
+      },
+      [blocksLayoutFieldname]: {
+        items: [
+          ...formData[blocksLayoutFieldname].items.slice(0, index),
+          idnewBlock,
```
Comment (nileshgulia1) [inline src/helpers/Blocks/Blocks.js]: Replacing the current block with the newer block created. I like this :)

**Comment by nileshgulia1** [comment]
Continued in #2337

**Pull request closed without merging**